api/admin/controller/discovery_services.py

```diff
@@ -93,6 +93,12 @@ def process_post(self):
             self._db.rollback()
             return name_error
 
+        url = flask.request.form.get("url")
+        url_error = self.check_url_unique(service, url, protocol, self.goal)
+        if url_error:
```

Comment:
Maybe call this something other than `url_error`?  That sounds like the errors we throw when a URL isn't in a valid format.